### PR TITLE
Add re-org env for Ethereum

### PR DESCRIPTION
**Pull request by skudasov**
Generic Helm deployment resource and Geth PoW network, without proof validation and with 1-3s block times for re-org testing:
```
NETWORK="ethereum_geth_reorg" ginkgo -r --focus="@runlog"
```

### CODE REVIEW

File: environment/k8s_environment_helm_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/smartcontractkit/integrations-framework/client"
+	"github.com/smartcontractkit/integrations-framework/config"
+	"github.com/smartcontractkit/integrations-framework/tools"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = XDescribe("Environment with geth re-org functionality @helm", func() {
```
Comment (tateexon) [inline environment/k8s_environment_helm_test.go:13]: Did you want this to still be a pending test?

File: contracts/ethereum_reorg_util.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	"context"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/smartcontractkit/integrations-framework/client"
+	"time"
+)
+
+const (
+	DAGAwaitTimeout = 120 * time.Second
```
Comment (skudasov) [inline contracts/ethereum_reorg_util.go:13]: Around 30s, depends on K8s resources.

File: environment/k8s_helm_resource.go [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+		LabelSelector: k.releaseSelector(),
+	})
+	if err != nil {
+		return nil, err
+	}
+	return svcList.Items, nil
+}
+
+func (k *K8sHelmManifest) ServiceDetails() ([]*ServiceDetails, error) {
+	sd := make([]*ServiceDetails, 0)
```
Comment (jleeh) [inline environment/k8s_helm_resource.go:161]: This function can just be the same as the original manfiest deployer? All it's doing is looping over the ports forwarded.

File: environment/k8s_helm_resource.go [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+const (
+	HelmInstallTimeout   = 200 * time.Second
+	ReleasePrefix        = "release"
+	DefaultK8sConfigPath = ".kube/config"
+)
+
+type SetValuesHelmFunc func(resource *K8sHelmManifest) error
+
+// K8sHelmManifest common helm manifest data
+type K8sHelmManifest struct {
```
Comment (jleeh) [inline environment/k8s_helm_resource.go:33]: Rename this to `HelmChart`

File: environment/k8s_helm_resource.go [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+	network           *config.NetworkConfig
+	namespace         *coreV1.Namespace
+	SetValuesHelmFunc SetValuesHelmFunc
+	// Deployment properties
+	ports        []portforward.ForwardedPort
+	values       map[string]interface{}
+	stopChannels []chan struct{}
+}
+
+func (k *K8sHelmManifest) Teardown() error {
```
Comment (jleeh) [inline environment/k8s_helm_resource.go:50]: This should close opened ports like the original deployer does.

File: environment/k8s_helm_resource.go [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+	return nil
+}
+
+// NewReorgGethHelmResource creates new helm manifest for multi-node Geth network
+func NewReorgGethHelmResource() *K8sHelmManifest {
+	return &K8sHelmManifest{
+		id:          "evm",
+		chartPath:   filepath.Join(tools.ProjectRoot, "environment/charts/geth-reorg"),
+		releaseName: "reorg-1",
+		values:      make(map[string]interface{}),
```
Comment (jleeh) [inline environment/k8s_helm_resource.go:64]: I don't see this being used?

File: environment/k8s_helm_resource.go [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+func (k *K8sHelmManifest) Teardown() error {
+	log.Debug().Str("Release", k.releaseName).Msg("Uninstalling Рelm release")
+	if _, err := action.NewUninstall(k.actionConfig).Run(k.releaseName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewReorgGethHelmResource creates new helm manifest for multi-node Geth network
+func NewReorgGethHelmResource() *K8sHelmManifest {
```
Comment (jleeh) [inline environment/k8s_helm_resource.go:59]: Rename to `NewGethReorgHelmChart`.

**Review by jleeh — CHANGES_REQUESTED** [review]
Overall, nice clean implementation of Helm.

**Review by jleeh — APPROVED** [review]

**Pull request merged**